lms/djangoapps/course_api/tests/test_views.py

```diff
@@ -0,0 +1,146 @@
+
+
+class CourseListViewTestCase(CourseApiTestViewMixin, SharedModuleStoreTestCase):
+    """
+    Test responses returned from CourseListView.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseListViewTestCase, cls).setUpClass()
+        cls.course = cls.create_course()
+        cls.url = reverse('course-list')
+        cls.staff_user = cls.create_user(username='staff', is_staff=True)
+        cls.honor_user = cls.create_user(username='honor', is_staff=False)
+
+    def test_as_staff(self):
+        self.setup_user(self.staff_user)
+        self.verify_response()
+
+    def test_as_staff_for_honor(self):
+        self.setup_user(self.staff_user)
+        self.verify_response(params={'username': self.honor_user.username})
+
+    def test_as_honor(self):
+        self.setup_user(self.honor_user)
+        self.verify_response()
+
+    def test_as_honor_for_explicit_self(self):
+        self.setup_user(self.honor_user)
+        self.verify_response(params={'username': self.honor_user.username})
+
+    def test_as_honor_for_staff(self):
+        self.setup_user(self.honor_user)
+        self.verify_response(expected_status_code=403, params={'username': self.staff_user.username})
+
+    def test_not_logged_in(self):
+        self.client.logout()
+        self.verify_response()
+
+
+class CourseDetailViewTestCase(CourseApiTestViewMixin, SharedModuleStoreTestCase):
+    """
+    Test responses returned from CourseDetailView.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseDetailViewTestCase, cls).setUpClass()
+        cls.course = cls.create_course()
+        cls.hidden_course = cls.create_course(course=u'hidden', visible_to_staff_only=True)
+        cls.url = reverse('course-detail', kwargs={'course_key_string': cls.course.id})
+        cls.hidden_url = reverse('course-detail', kwargs={'course_key_string': cls.hidden_course.id})
+        cls.nonexistent_url = reverse('course-detail', kwargs={'course_key_string': 'edX/nope/Fall_2014'})
+        cls.staff_user = cls.create_user(username='staff', is_staff=True)
+        cls.honor_user = cls.create_user(username='honor', is_staff=False)
+
+    def test_as_honor(self):
+        self.setup_user(self.honor_user)
+        self.verify_response()
+
+    def test_as_honor_for_explicit_self(self):
+        self.setup_user(self.honor_user)
+        self.verify_response(params={'username': self.honor_user.username})
+
+    def test_as_honor_for_staff(self):
+        self.setup_user(self.honor_user)
+        self.verify_response(expected_status_code=403, params={'username': self.staff_user.username})
+
+    def test_as_staff(self):
+        self.setup_user(self.staff_user)
+        self.verify_response()
+
+    def test_as_staff_for_honor(self):
+        self.setup_user(self.staff_user)
+        self.verify_response(params={'username': self.honor_user.username})
+
+    def test_as_anonymous_user(self):
+        self.verify_response(expected_status_code=401)
```

Comment:
Curious: I thought the anonymous user had 'see_exists' permission to the course.  No?